packages/manager/modules/cloud-connect/src/details/statistics/statistics.controller.js

```diff
@@ -0,0 +1,216 @@
+        break;
+      default:
+        this.loadTrafficGraph();
+        this.displayedGraph = this.TYPE.TRAFFIC;
+        break;
+    }
+  }
+
+  // Load error statistics
+  loadErrorGraph() {
+    this.series = [];
+    this.data = [];
+    this.isLoading = true;
+
+    // Update options
+    this.options.scales.yAxes[0].scaleLabel.labelString = this.$translate.instant(
+      'cloud_connect_stats_error_unit',
+    );
+
+    this.cloudConnect.interfaceList.forEach((interfaceId) => {
+      return this.$q
+        .all({
+          download: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.ERROR_DOWN,
+            this.graphPeriod,
+          ),
+          upload: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.ERROR_UP,
+            this.graphPeriod,
+          ),
+        })
+        .then((stats) => {
+          if (stats.download.length > 0 || stats.upload.length > 0) {
+            this.updateStats(this.TYPE.ERROR, interfaceId, stats);
+            this.isLoading = false;
+          }
+        });
+    });
+  }
+
+  // Load light statistics
+  loadLightGraph() {
+    this.series = [];
+    this.data = [];
+    this.isLoading = true;
+
+    this.cloudConnect.interfaceList.forEach((interfaceId) => {
+      return this.$q
+        .all({
+          in: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.LIGHT_IN,
+            this.graphPeriod,
+          ),
+          out: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.LIGHT_OUT,
+            this.graphPeriod,
+          ),
+        })
+        .then((stats) => {
+          if (stats.in.length > 0 || stats.out.length > 0) {
+            this.updateStats(this.TYPE.LIGHT, interfaceId, stats);
+            this.isLoading = false;
+          }
+        });
+    });
+  }
+
+  // Load traffic statistics
+  loadTrafficGraph() {
+    this.series = [];
+    this.data = [];
+    this.isLoading = true;
+
+    // Update options
+    this.options.scales.yAxes[0].scaleLabel.labelString = this.$translate.instant(
+      'cloud_connect_stats_traffic_unit',
+    );
+
+    this.cloudConnect.interfaceList.forEach((interfaceId) => {
+      return this.$q
+        .all({
+          download: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.TRAFFIC_DOWN,
+            this.graphPeriod,
+          ),
+          upload: this.loadStatistics(
+            this.cloudConnect.id,
+            interfaceId,
+            this.TYPE.TRAFFIC_UP,
+            this.graphPeriod,
+          ),
+        })
+        .then((stats) => {
+          if (stats.download.length > 0 || stats.upload.length > 0) {
+            this.updateStats(this.TYPE.TRAFFIC, interfaceId, stats);
+            this.isLoading = false;
+          }
+        });
+    });
```

Comment:
Looks like the same as l113, maybe we could create a function for this purpose

Reply:
No it's the same role

Reply:
And could we have a fucntion that takes the role as a parameter to avoid duplicating what looks really similar